osu.Game/Screens/Edit/Screens/Setup/Components/LabelledComponents/LabelledTextBox.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using static osu.Framework.Graphics.UserInterface.TextBox;
+
+namespace osu.Game.Screens.Edit.Screens.Setup.Components.LabelledComponents
+{
+    public class LabelledTextBox : CompositeDrawable
+    {
+        private readonly OsuTextBox textBox;
+        private readonly Container content;
+        private readonly OsuSpriteText label;
+
+        private const float label_container_width = 150;
+        private const float outer_corner_radius = 15;
+        private const float inner_corner_radius = 10;
+        private const float default_height = 40;
+        private const float default_label_left_padding = 15;
+        private const float default_label_top_padding = 12;
+        private const float default_label_text_size = 16;
+
+        public event OnCommitHandler OnCommit;
+
+        private bool readOnly;
```

Comment:
For all these properties - just use the underlying value (e.g. textBox.ReadOnly).